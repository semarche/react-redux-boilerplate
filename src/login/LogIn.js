import React, { Component } from 'react';
import { Redirect } from 'react-router-dom';
import PropTypes from 'prop-types';
import Button from '../ui_parts/Button';
import './style.scss';

const propTypes = {
  isAuthenticated: PropTypes.bool.isRequired,
  authentication: PropTypes.func.isRequired,
};

class LogIn extends Component {
  render() {
    const { isAuthenticated, authentication } = this.props;
    return (isAuthenticated ? <Redirect to={'/'} /> : (
        <div className='login-form'>
          <h2>LogIn form</h2>
          <h4>Click <Button text={'THERE'} onClick={authentication} /> for Log in</h4>
        </div>
      )
    )
  }
}

LogIn.propTypes = propTypes;
export default LogIn;
