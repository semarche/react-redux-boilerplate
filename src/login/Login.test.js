import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import LogIn  from './LogIn';

configure({ adapter: new Adapter() });

describe('Login', () => {
  const wrapper = shallow(
    <LogIn
      authentication={jest.fn()}
      isAuthenticated={false}
    />
  )

  it('should render LogIn component', () => {
    expect(wrapper.find('.login-form').exists()).toEqual(true);
  });
  it('matches the snapshot', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
