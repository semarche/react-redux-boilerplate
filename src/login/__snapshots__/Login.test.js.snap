// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login matches the snapshot 1`] = `
<div
  className="login-form"
>
  <h2>
    LogIn form
  </h2>
  <h4>
    Click 
    <Button
      onClick={[MockFunction]}
      text="THERE"
    />
     for Log in
  </h4>
</div>
`;
